src/main/java/de/tum/in/www1/artemis/service/ProgrammingSubmissionService.java

```diff
@@ -182,11 +183,16 @@ public ProgrammingSubmission notifyPush(Long participationId, Object requestBody
      * Trigger the CI of all student participations of the given exercise.
      * The build result will become rated regardless of the due date as the submission type is INSTRUCTOR.
      *
+     * The method is async because it would timeout a calling resource method.
```

Comment:
Why would this lead to a timeout? I thought this only triggers the builds, but does not wait until they come in, so this should be quite fast even for thousands of students.
And as far as I know, the timeouts are really long. I remember, when I archived repos once that it took more than 10minutes until the response to the client was sent, and there was still no timeout.

To be honest: I am not a big fan of async, because if something goes wrong the client might not notice it....

Reply:
If the service method is not async, everything will be executed in the same thread, so the response will only be sent out when all submissions are created and all builds have been triggered on Bamboo.
I think this is a good use case for async.